test/OrchardCore.Tests/Recipes/RecipeExecutorTests.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+using Moq;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Builders;
+using OrchardCore.Environment.Shell.Scope;
+using OrchardCore.Recipes.Events;
+using OrchardCore.Recipes.Models;
+using OrchardCore.Recipes.Services;
+using OrchardCore.Scripting;
+using Xunit;
+
+namespace OrchardCore.Recipes
+{
+    public class RecipeExecutorTests
+    {
+        [Theory]
+        [InlineData("recipe1", "[locale en]You have successfully registered![/locale][locale fr]Vous vous êtes inscrit avec succès![/locale]")]
+        [InlineData("recipe2", "[1js: valiables('now')]")]
+        [InlineData("recipe3", "js: valiables('now')")]
+        [InlineData("recipe4", "[locale en]This text contain a colon ':' symbol[/locale][locale fr]Ce texte contient un symbole du côlon ':'[/locale]")]
+        [InlineData("recipe5", "[sc text='some : text'/]")]
+        public async Task ShouldTrimValidScriptExpression(string recipeName, string expected)
+        {
+            // Arrange
+            var shellHostMock = new Mock<IShellHost>();
+            var shellSettings = new ShellSettings { Name = "Test" };
+            var services = new ServiceCollection()
+                .AddScripting();
+            shellHostMock.Setup(h => h.GetScopeAsync(It.IsAny<ShellSettings>()))
```

Comment:
i suspect it's how you're building the shellhost mock here. I'll have a look tomorrow

Reply:
Yeah this isn't going to work as a mock, because the ShellScope is a static asynclock.

Have a look at how @jtkech rebuilt the unit tests for the `AutorouteEntriesTests` when that moved to needing a `ShellScope`

Reply:
I tried to make the code looks like

```csharp
var shellContext = CreateShellContext();
await shellContext.CreateScope().UsingAsync(async scope =>
{
    // Arrange
    var shellHostMock = new Mock<IShellHost>();
    shellHostMock.Setup(h => h.GetScopeAsync(It.IsAny<ShellSettings>())).Returns(Task.FromResult(scope));

    var recipeEventHandlers = new List<IRecipeEventHandler> { new RecipeEventHandler() };
    var loggerMock = new Mock<ILogger<RecipeExecutor>>();
    var recipeExecutor = new RecipeExecutor(shellHostMock.Object, shellContext.Settings, recipeEventHandlers, loggerMock.Object);
    var executionId = Guid.NewGuid().ToString("n");
    var recipeDescriptor = new RecipeDescriptor { RecipeFileInfo = GetRecipeFileInfo(recipeName) };

    // Act
    await recipeExecutor.ExecuteAsync(executionId, recipeDescriptor, new object(), CancellationToken.None);

    // Assert
    var recipeContext = (recipeEventHandlers.Single() as RecipeEventHandler).Context;
    var recipeStep = recipeContext.Step;
    Assert.Equal(expected, recipeStep.SelectToken("data.[0].TitlePart.Title").ToString());
});
```

but still the tests fail :( @jtkech any idea?

Reply:
@hishamco @deanmarcussen 

Okay, i will take a look on it this week end